src/database/type-transforms.tsx

```diff
@@ -109,3 +111,67 @@ export const buildUser = (document: firebase.firestore.DocumentData): User => {
 
   return { ...pick(data, ["name", "email"]), id: document.id };
 };
+
+const buildCovidCase = (
+  doc: firebase.firestore.DocumentData,
+): ShadowCovidCase => {
+  const data = doc.data();
+
+  const observedAt = parse(doc.id, "yyyy-MM-dd", startOfToday());
+
+  return {
+    observedAt,
+    ...data,
+  };
+};
+
+export const buildShadowFacility = (
+  facilityDocument: firebase.firestore.DocumentData,
+  facilityCovidCaseDocuments: firebase.firestore.DocumentData[],
+): ShadowFacility => {
+  const data = facilityDocument.data();
+
+  // convert all known timestamps to dates
+  const toSimpleTimeseries = (record: {
+    date: firebase.firestore.Timestamp;
+    value: number;
+  }) => ({
+    date: timestampToDate(record.date),
+    value: record.value,
+  });
+
+  let {
+    state,
+    canonicalName,
+    facilityType,
+    capacity,
+    population,
+    ...other
+  } = data;
+
+  // cast known types
+  state = String(state);
+  canonicalName = String(canonicalName);
+  facilityType = String(facilityType);
+  // if these are not arrays then unfortunately they are garbage
+  if (!Array.isArray(capacity)) {
+    capacity = [];
+  }
+  if (!Array.isArray(population)) {
+    population = [];
+  }
+
+  capacity = capacity.map(toSimpleTimeseries);
+  population = population.map(toSimpleTimeseries);
+
+  return {
+    id: facilityDocument.id,
+    state,
+    canonicalName,
+    facilityType,
+    capacity,
+    population,
+    covidCases: facilityCovidCaseDocuments.map(buildCovidCase),
```

Comment:
This is _extreme_ paranoia, but it may be good practice anyway: if we happen to be performing this `map` call while we pass over the midnight line, given that the internal call to `buildCovidCase` invokes `startOfToday()`, we'd have inconsistent dates on some of the results in the collection. It might be wise to pass a date into the function to use as `observedAt` and then initialize that once out here.